# PERGUNTA: Porcentagem de conflitos que aconteceram depois do aquisição da bomba nuclear pelo país

In [1]:
import pandas as pd

In [ ]:
# !pip install basedosdados
# !pip install matplotlib
# !pip install pandas

In [2]:
nuclear_stockpiles = pd.read_csv('nuclear_weapon/nuclear_weapons_stockpiles.csv', delimiter=',')

In [3]:
conflicts = pd.read_csv('conflicts_and_wars_202403110030.csv', delimiter=',')

In [ ]:
conflicts

In [4]:
display(conflicts.tail(),nuclear_stockpiles.tail())

Date                                           Headline Conflict Type  \
922  2019  One Israeli killed and two others injured at a...       Trigger   
923  2019  50,000 people flee their homes in central Mali...       Trigger   
924  2019  Violence in northwest Syria damages civilian i...      Casualty   
925  2019  Shrapnel damages a water pipeline, leaving mor...      Casualty   
926  2022                                Conflict in Ukraine        Weapon   

          Country              Region  \
922     Palestine        Western Asia   
923          Mali  Sub-Saharan Africa   
924         Syria        Western Asia   
925       Ukraine      Eastern Europe   
926  Keiv,Ukraine      Eastern Europe   

                                           Description  \
922  A homemade bomb explodes at a spring, killing ...   
923  A series of massacres in central Mali, fueled ...   
924  From January through July 2019 violence in nor...   
925  Shrapnel from an exploding shells damages a pi...   
926  The Russo-Ukrainian War is an ongoing war prim...   

                                          Sources  
922                   Chronicle News Service 2019  
923                                     Webb 2019  
924  US Agency for International Development 2019  
925                                   UNICEF 2019  
926                                   UNICEF 2020

country_name  year  nuclear_weapons_stockpile
775  United States  2018                       3785
776  United States  2019                       3805
777  United States  2020                       3750
778  United States  2021                       3708
779  United States  2022                       3708

In [5]:
country_to_region = dict(zip(conflicts['Country'], conflicts['Region']))

In [6]:
nuclear_stockpiles['Region'] = nuclear_stockpiles['country_name'].map(country_to_region)
nuclear_stockpiles.tail()

country_name  year  nuclear_weapons_stockpile         Region
775  United States  2018                       3785  North America
776  United States  2019                       3805  North America
777  United States  2020                       3750  North America
778  United States  2021                       3708  North America
779  United States  2022                       3708  North America

In [7]:
nuclear_stockpiles

country_name  year  nuclear_weapons_stockpile         Region
0            China  1945                          0   Eastern Asia
1            China  1946                          0   Eastern Asia
2            China  1947                          0   Eastern Asia
3            China  1948                          0   Eastern Asia
4            China  1949                          0   Eastern Asia
..             ...   ...                        ...            ...
775  United States  2018                       3785  North America
776  United States  2019                       3805  North America
777  United States  2020                       3750  North America
778  United States  2021                       3708  North America
779  United States  2022                       3708  North America

[780 rows x 4 columns]

In [8]:
nuclear_stockpiles[nuclear_stockpiles["nuclear_weapons_stockpile"] != 0]

country_name  year  nuclear_weapons_stockpile         Region
19           China  1964                          1   Eastern Asia
20           China  1965                          5   Eastern Asia
21           China  1966                         20   Eastern Asia
22           China  1967                         25   Eastern Asia
23           China  1968                         35   Eastern Asia
..             ...   ...                        ...            ...
775  United States  2018                       3785  North America
776  United States  2019                       3805  North America
777  United States  2020                       3750  North America
778  United States  2021                       3708  North America
779  United States  2022                       3708  North America

[462 rows x 4 columns]

In [9]:
nuclear_nonzero = nuclear_stockpiles[nuclear_stockpiles["nuclear_weapons_stockpile"] != 0]

In [40]:
periodo_pais_tem_arma_nuclear = nuclear_nonzero.groupby("country_name")["year"].agg(['first', 'last'])
periodo_pais_tem_arma_nuclear.head()

first  last
country_name             
China          1964  2022
France         1964  2022
India          1998  2022
Israel         1967  2022
North Korea    2015  2022

In [32]:
#Primeiro: pega o primeiro pais da lista de 'periodo_pais_tem_arma_nuclear'
#Segundo: Retira todos os anos com 'BC' e 'AD'
#Terceiro: Retira todos os confitos com year1-year2 e transforma apenas para year1
#Quarto: pega todos as datas com um numero maior ou igual do que o primeiro ano de 'periodo_pais_tem_arma_nuclear' e menor e igual ultimo ano de 'periodo_pais_tem_arma_nuclear'


chinaConflicts = conflicts[conflicts["Country"].str.contains('China')]
chinaConflicts = chinaConflicts[~chinaConflicts['Date'].str.endswith(('BC', 'AD'))]
chinaConflicts['Date'] = chinaConflicts['Date'].str.slice(0, 4)
chinaConflicts['Date'] = chinaConflicts['Date'].astype(int)
chinaConflicts = chinaConflicts[(chinaConflicts["Date"] >= periodo_pais_tem_arma_nuclear.values[0][0]) & (chinaConflicts["Date"] <= periodo_pais_tem_arma_nuclear.values[0][1])]
chinaConflicts

Date                                           Headline  \
104  1970  Violent conflicts over use of China's Zhang River   
117  1976  Militia Chief shot in China's Zhang River conf...   
167  1991  Chinese villages exhange mortar fire over wate...   
174  1992                Canal on China's Zhang River bombed   
177  1992  Dispute over control of Zhang River, China lea...   
220  1999                   Chinese farmers fight over water   
248  2000                            Riots in northern China   
264  2001                     Chinese protestors block canal   
329  2003  A man poisons a drinking water reservoir in He...   
341  2004                 Police kill dam protestor in China   
342  2004         Taiwan to target China's water in defense?   
381  2006                    Dam protester executed in China   
432  2008                         China cracks down on Tibet   
529  2012  Chinese protestors challenge Japanese factory ...   
559  2013  Chinese dam builders killed after attack by Uy...   

         Conflict Type Country         Region  \
104            Trigger   China   Eastern Asia   
117            Trigger   China   Eastern Asia   
167            Trigger   China   Eastern Asia   
174  Trigger, Casualty   China   Eastern Asia   
177  Trigger, Casualty   China   Eastern Asia   
220            Trigger   China   Eastern Asia   
248  Trigger, Casualty   China   Eastern Asia   
264    Trigger, Weapon   China   Eastern Asia   
329             Weapon   China  Eastern China   
341            Trigger   China   Eastern Asia   
342           Casualty   China   Eastern Asia   
381            Trigger   China   Eastern Asia   
432            Trigger   China   Eastern Asia   
529            Trigger   China   Eastern Asia   
559  Trigger, Casualty   China   Eastern Asia   

                                           Description  \
104  Conflicts over excessive water withdrawals and...   
117  A local militia chief is shot to death in a cl...   
167  In December 1991, Huanglongkou village and Qia...   
174  In August 1992, bombs are set off along a Zhan...   
177  In August 1992, farmers of Baishan village in ...   
220  Around the Chinese New Year, farmers from Hebe...   
248  Civil unrest erupts over use and allocation of...   
264  In an act to protest the destruction of fisher...   
329  A man who sells water purifiers poisons a rese...   
341  Tens of thousands of farmers stage a sit-in ag...   
342  A 2004 Pentagon report on China's military cap...   
381  Chinese authorities execute a man who took par...   
432  China launches a political crackdown in Tibet....   
529  In July, thousands of protesters take to the s...   
559  Hundreds of Uyghurs are arrested following an ...   

                                          Sources  
104              China Water Resources Daily 2002  
117              China Water Resources Daily 2002  
167              China Water Resources Daily 2002  
174              China Water Resources Daily 2002  
177                                    Pietz 2015  
220              China Water Resources Daily 2002  
248                                Pottinger 2000  
264        China Ministry of Water Resources 2001  
329                      Mohtadi and Murshid 2006  
341                                      Lim 2004  
342                             China Daily 2004a  
381  British Broadcasting Corporation (BBC) 2006d  
432                                  Sharife 2008  
529                                  Economy 2013  
559                                   Hoshur 2013

In [41]:
#Primeiro: pega o primeiro pais da lista de 'periodo_pais_tem_arma_nuclear'
#Segundo: Retira todos os anos com 'BC' e 'AD'
#Terceiro: Retira todos os confitos com year1-year2 e transforma apenas para year1
#Quarto: pega todos as datas com um numero maior ou igual do que o primeiro ano de 'periodo_pais_tem_arma_nuclear' e menor e igual ultimo ano de 'periodo_pais_tem_arma_nuclear'


FranceConflicts = conflicts[conflicts["Country"].str.contains('France')]
FranceConflicts = FranceConflicts[~FranceConflicts['Date'].str.endswith(('BC', 'AD'))]
FranceConflicts['Date'] = FranceConflicts['Date'].str.slice(0, 4)
FranceConflicts['Date'] = FranceConflicts['Date'].astype(int)
FranceConflicts = FranceConflicts[(FranceConflicts["Date"] >= periodo_pais_tem_arma_nuclear.values[1][0]) & (FranceConflicts["Date"] <= periodo_pais_tem_arma_nuclear.values[1][1])]
FranceConflicts

Date                                           Headline Conflict Type  \
228  1999        Water supply pipeline is attacked in France      Casualty   
232  1999  Multiple attacks on utilities, including water...      Casualty   
233  1999  Bomb explodes on an island, cutting off water ...      Casualty   
242  2000    French workers pollute river over labor dispute      Casualty   
251  2000  Water company is bombed to demand release of p...      Casualty   
303  2002     Bomb explodes in water utility office building      Casualty   

    Country          Region  \
228  France  Western Europe   
232  France  Western Europe   
233  France  Western Europe   
242  France  Western Europe   
251  France  Western Europe   
303  France  Western Europe   

                                           Description  \
228  Explosives are placed along a water supply pip...   
232  Reports find at least three attacks on utiliti...   
233  A bomb explodes on the island of Cavallo, off ...   
242  In July, workers at the Cellatex chemical plan...   
251  A water company office in Ustaritz, France is ...   
303  A powerful bomb explodes in a water utility of...   

                                               Sources  
228                    France-Inter Radio Network 1999  
232  RAND Database of Worldwide Terrorism Incidents...  
233  RAND Database of Worldwide Terrorism Incidents...  
242                     Christian Science Monitor 2000  
251  RAND Database of Worldwide Terrorism Incidents...  
303  RAND Database of Worldwide Terrorism Incidents...

In [57]:
conflicts[conflicts["Country"].str.contains('France')].count().values[0]

8

In [59]:
#Primeiro: pega o primeiro pais da lista de 'periodo_pais_tem_arma_nuclear'
#Segundo: Retira todos os anos com 'BC' e 'AD'
#Terceiro: Retira todos os confitos com year1-year2 e transforma apenas para year1
#Quarto: pega todos as datas com um numero maior ou igual do que o primeiro ano de 'periodo_pais_tem_arma_nuclear' e menor e igual ultimo ano de 'periodo_pais_tem_arma_nuclear'


IndiaConflicts = conflicts[conflicts["Country"].str.contains('India')]
IndiaConflicts = IndiaConflicts[~IndiaConflicts['Date'].str.endswith(('BC', 'AD'))]
IndiaConflicts['Date'] = IndiaConflicts['Date'].str.slice(0, 4)
IndiaConflicts['Date'] = IndiaConflicts['Date'].astype(int)
IndiaConflicts = IndiaConflicts[(IndiaConflicts["Date"] >= periodo_pais_tem_arma_nuclear.values[2][0]) & (IndiaConflicts["Date"] <= periodo_pais_tem_arma_nuclear.values[2][1])]
IndiaConflicts

Date                                           Headline  \
224  1999                      Water riots in Gujarat, India   
281  2002                     Deadly clash in Kashmir, India   
284  2002               Violence over Cauvery River in India   
289  2002  Indian military protecting reservoir attacked ...   
325  2003  Group bombs a rural water supply office and ot...   
340  2004                Terrorists bomb water pipe in India   
345  2004              Police kill water protestors in India   
395  2006  200 people attack home of Water Resource Minister   
400  2006      Two bombs explode at a water collection point   
409  2007               Injuries at protest in Orissa, India   
452  2009  Drought and water inequities spark killings India   
453  2009       Man killed in Mumbai protest over water cuts   
470  2010                     Violent water protest in India   
472  2010                   Protestors killed and injured in   
479  2010  Militants blow up pump houses belonging to the...   
496  2011        Unidentified militants blow up a water pump   
504  2012  Injuries during protest over dam releases in I...   
505  2012  Violent protests over water shortages in New D...   
506  2012  Pakistani militants attack water systems in Ka...   
531  2012  Militants block work at the Wullar Dam constru...   
534  2012  Women assaulted for attempting to take well wa...   
557  2013    Water shortage leads to local fighting in India   
568  2013      Pakistan and India clash over Siachen Glacier   
578  2013                Water tanker in Myanmar is attacked   
595  2014  Upper caste women reportedly restrict access t...   
601  2014  Armed bandits force villages to pay a 'water tax'   
624  2014               Water construction equipment damaged   
687  2016  Scarce water supplies in drought regions of In...   
690  2016  Protests against the building of dams on the B...   
691  2016               Riots over water in Bangalore, India   
694  2016  18 killed and 200 injured after reopening of t...   
696  2016  Armed gaurds clash with farmers over drought-s...   
708  2016                     Attack on a local dam in India   
712  2016  Water shortage led to clashes between two grou...   
820  2017  Police kill five farmers during protests over ...   
831  2018  Water shortages in India lead to deadly fights...   
898  2019  One death reported in a fight over water in so...   
899  2019  One man stabs six others in a fight over water...   
901  2019  India responds to deadly attack by threatening...   
902  2019  A man is stabbed by his neighbor over a water ...   
908  2019  A grandmother and her grandson are beaten when...   
913  2019  Numerous fights, including attacks on water ta...   
914  2019  Two men are seriously injured in clash over ac...   

         Conflict Type          Country         Region  \
224            Trigger            India  Southern Asia   
281            Trigger            India  Southern Asia   
284            Trigger            India  Southern Asia   
289           Casualty            India  Southern Asia   
325           Casualty            India  Southern Asia   
340           Casualty            India  Southern Asia   
345            Trigger            India  Southern Asia   
395           Casualty            India  Southern Asia   
400           Casualty            India  Southern Asia   
409            Trigger            India  Southern Asia   
452            Trigger            India  Southern Asia   
453            Trigger            India  Southern Asia   
470            Trigger            India  Southern Asia   
472            Trigger            India  Southern Asia   
479           Casualty            India  Southern Asia   
496           Casualty            India  Southern Asia   
504            Trigger            India  Southern Asia   
505            Trigger            India  Southern Asia   
506  Trigger, Casualty  India, Pakistan  Southern Asia   
531            Trigger  Indi

In [60]:
#Primeiro: pega o primeiro pais da lista de 'periodo_pais_tem_arma_nuclear'
#Segundo: Retira todos os anos com 'BC' e 'AD'
#Terceiro: Retira todos os confitos com year1-year2 e transforma apenas para year1
#Quarto: pega todos as datas com um numero maior ou igual do que o primeiro ano de 'periodo_pais_tem_arma_nuclear' e menor e igual ultimo ano de 'periodo_pais_tem_arma_nuclear'


IsraelConflicts = conflicts[conflicts["Country"].str.contains('Israel')]
IsraelConflicts = IsraelConflicts[~IsraelConflicts['Date'].str.endswith(('BC', 'AD'))]
IsraelConflicts['Date'] = IsraelConflicts['Date'].str.slice(0, 4)
IsraelConflicts['Date'] = IsraelConflicts['Date'].astype(int)
IsraelConflicts = IsraelConflicts[(IsraelConflicts["Date"] >= periodo_pais_tem_arma_nuclear.values[3][0]) & (IsraelConflicts["Date"] <= periodo_pais_tem_arma_nuclear.values[3][1])]
IsraelConflicts

Date                                           Headline  \
97   1967                  Israel attacks Jordan water works   
101  1969                      Israel destroys Jordan canals   
138  1983             Terrorists plot to poison Israel water   
155  1989  Bomb discovered near a water pumping station p...   
205  1998  Water line attack leaves 11 communities withou...   
261  2001  Palestinians destroy water supply to Israeli s...   
379  2005  Rocket lands near an Israeli national water co...   
385  2006  Lebanon, Israel target water infrastructure in...   
405  2006  Rocket attack damages a Mekorot water station,...   
487  2011  Israel destroys Palestinians' water pumps and ...   
488  2011  Israeli settlers destroy wells in attack on We...   

        Conflict Type            Country        Region  \
97           Casualty     Israel, Jordan  Western Asia   
101          Casualty     Israel, Jordan  Western Asia   
138            Weapon             Israel  Western Asia   
155  Casualty, Weapon             Israel  Western Asia   
205          Casualty             Israel  Western Asia   
261          Casualty  Israel, Palestine  Western Asia   
379          Casualty             Israel  Western Asia   
385          Casualty    Israel, Lebanon  Western Asia   
405          Casualty             Israel  Western Asia   
487          Casualty  Palestine,?Israel  Western Asia   
488          Casualty  Palestine, Israel  Western Asia   

                                           Description  \
97   Israel destroys the Arab diversion works on th...   
101  Israel, suspicious that Jordan is over-diverti...   
138  The Israeli government reports that it had unc...   
155  A large amount of dynamite is discovered near ...   
205  Eleven communities in the Negev area of Israel...   
261  Palestinians destroy water supply pipelines to...   
379  In one of two incidents in Israel on this day ...   
385  Hezbollah rockets damage a wastewater plant in...   
405  Palestinian Islamic Jihad (PIJ) claims respons...   
487  Israel's military destroys nine water tanks in...   
488  Israelis from nearby settlements attack Qasra,...   

                                               Sources  
97                                         Gleick 1993  
101                           Samson and Charrier 1997  
138                      Douglass and Livingstone 1987  
155  RAND Database of Worldwide Terrorism Incidents...  
205  RAND Database of Worldwide Terrorism Incidents...  
261                                  Israel Line 2001a  
379  National Consortium for the Study of Terrorism...  
385                                       Science 2006  
405  National Consortium for the Study of Terrorism...  
487                                       Aburawa 2011  
488                           Bsharat and Ramadan 2011

In [61]:
#Primeiro: pega o primeiro pais da lista de 'periodo_pais_tem_arma_nuclear'
#Segundo: Retira todos os anos com 'BC' e 'AD'
#Terceiro: Retira todos os confitos com year1-year2 e transforma apenas para year1
#Quarto: pega todos as datas com um numero maior ou igual do que o primeiro ano de 'periodo_pais_tem_arma_nuclear' e menor e igual ultimo ano de 'periodo_pais_tem_arma_nuclear'


NorthKoreaConflicts = conflicts[conflicts["Country"].str.contains('North Korea')]
NorthKoreaConflicts = NorthKoreaConflicts[~NorthKoreaConflicts['Date'].str.endswith(('BC', 'AD'))]
NorthKoreaConflicts['Date'] = NorthKoreaConflicts['Date'].str.slice(0, 4)
NorthKoreaConflicts['Date'] = NorthKoreaConflicts['Date'].astype(int)
NorthKoreaConflicts = NorthKoreaConflicts[(NorthKoreaConflicts["Date"] >= periodo_pais_tem_arma_nuclear.values[4][0]) & (NorthKoreaConflicts["Date"] <= periodo_pais_tem_arma_nuclear.values[4][1])]
NorthKoreaConflicts

Empty DataFrame
Columns: [Date, Headline, Conflict Type, Country, Region, Description, Sources]
Index: []

In [54]:
PorcentagemDeConflitosAposConseguirArmaNuclear = []

In [55]:
resposta = chinaConflicts.count().values[0]/conflicts[conflicts["Country"].str.contains('China')].count().values[0]
PorcentagemDeConflitosAposConseguirArmaNuclear.append(resposta * 100)
PorcentagemDeConflitosAposConseguirArmaNuclear

[71.42857142857143]

In [58]:
resposta = FranceConflicts.count().values[0]/conflicts[conflicts["Country"].str.contains('France')].count().values[0]
PorcentagemDeConflitosAposConseguirArmaNuclear.append(resposta * 100)
PorcentagemDeConflitosAposConseguirArmaNuclear

[71.42857142857143, 75.0]

In [62]:
resposta = IndiaConflicts.count().values[0]/conflicts[conflicts["Country"].str.contains('India')].count().values[0]
PorcentagemDeConflitosAposConseguirArmaNuclear.append(resposta * 100)
PorcentagemDeConflitosAposConseguirArmaNuclear

[71.42857142857143, 75.0, 93.47826086956522]

In [63]:
resposta = IsraelConflicts.count().values[0]/conflicts[conflicts["Country"].str.contains('Israel')].count().values[0]
PorcentagemDeConflitosAposConseguirArmaNuclear.append(resposta * 100)
PorcentagemDeConflitosAposConseguirArmaNuclear

[71.42857142857143, 75.0, 93.47826086956522, 57.89473684210527]

In [64]:
resposta = NorthKoreaConflicts.count().values[0]/conflicts[conflicts["Country"].str.contains('North Korea')].count().values[0]
PorcentagemDeConflitosAposConseguirArmaNuclear.append(resposta * 100)
PorcentagemDeConflitosAposConseguirArmaNuclear

[71.42857142857143, 75.0, 93.47826086956522, 57.89473684210527, 0.0]

Porcentagem de conflitos que aconteceram depois do aquisição da bomba nuclear pelo pais, na order CHINA, FRANÇA, INDIA, ISRAEL, KOREA DO NORTE:

VALORES TRUNCADOS!


[71.42%, 75.0%, 93.47%, 57.89%, 0.0%]